### PR TITLE
refactor: use simple-git for git commands

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR implements:
- guardAgainstDirtyRepo only in CI mode
- moved __ CRAWLER_MODE __ checks into a function
- replaced all custom git commands with simple-git
- proxied the dangerous git actions with a condition 
- removed `git` and `exec` functions

### CODE REVIEW

File: packages/deprecation-crawler/src/lib/utils.ts [hunk 1]
```diff
@@ -184,26 +217,28 @@ export async function getCurrentHeadName(): Promise<string> {
...
     .map((tag) => tag.trim())
     .filter(Boolean);
 }
 
 export async function getCurrentBranchOrTag() {
   return (
-    (await git(['branch', '--show-current'])) ||
-    (await git(['describe', ' --tags --exact-match']))
+    (await git.branch().then((r) => r.current)) ||
+    // @TODO rplace with simple git
```
Comment (BioPhoton) [inline packages/deprecation-crawler/src/lib/utils.ts:229]:
```suggestion
    // @TODO replace with simple git
```

File: packages/deprecation-crawler/src/lib/deprecation-crawler.ts [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
 import { updateRepository } from './tasks/update-repository';
 import { addGroups } from './tasks/add-groups';
 import { generateOutput } from './tasks/generate-output';
 import { commitChanges } from './tasks/commit-changes';
-import { CRAWLER_MODES } from './constants';
 
 (async () => {
-  await guardAgainstDirtyRepo();
-
+  if (isCrawlerModeCi()) {
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/deprecation-crawler.ts:19]: We now have 2 checks for `guardAgainstDirtyRepo`.
This new one, and the existing `if (isCrawlerModeSandbox()) {` inside the `guardAgainstDirtyRepo` method.
Can you remove one of the conditions please?

**Review by timdeschryver — COMMENTED** [review]
Cool, I didn't know about `simple-git`, seems to be that it's easy to use.

**Comment by BioPhoton** [comment]
Will remove the double check.

Also the simple git lib is really neat for getting the string output to structured objects. 😁

Though it helps for our git diff crawling...

File: packages/deprecation-crawler/src/lib/deprecation-crawler.ts (hunk 2 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/deprecation-crawler.ts:19]: This is very interesting as it forces me to think about a standardized way of where to put "feature toggles"...
ATM I prefer to have the behavior e.g. `guardAgainstDirtyRepo` isolated in a function and the usage separate.
This obviously leads to repetitive checks when using it. :D

Opinions?

Comment (timdeschryver) [reply packages/deprecation-crawler/src/lib/deprecation-crawler.ts:19]: 👍 we can do that.
If we got a better idea later, we can always refactor 😅

**Pull request merged**